components/logins/src/sync/engine.rs

```diff
@@ -504,6 +506,22 @@ impl SyncEngine for LoginsSyncEngine {
     }
 }
 
+// This seems misplaced - but everywhere else a mirror row is read we don't need the unknown
+// fields because they are, um, unknown :) So we never merge or inspect them, just send
+// whatever we have in outgoing.
+fn unknown_fields_from_row(
+    row: &rusqlite::Row<'_>,
+) -> Result<Option<super::payload::UnknownFields>> {
+    Ok(match row.get::<_, Option<String>>("unknown_fields")? {
+        None => None,
+        Some(s) => match serde_json::from_str(&s)? {
+            serde_json::Value::Object(m) => Some(m),
+            // ???? Should maybe log/report an error?
+            _ => None,
+        },
+    })
+}
```

Comment:
nit: perhaps something like this can help to avoid nested matches?

```suggestion
match row.get::<_, Option<String>>("unknown_fields")?.map(serde_json::from_str) {
    serde_json::Value::Object(m) => Ok(m),
    _ => Ok(None),
},

```

Reply:
As discussed above though, in one scenario the None is expected, but the other needs to make lots of noise (and as mentioned, actually think a panic is OK)